File: frontend/demo/component/custom-field/custom-field-whitespace-theme.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/custom-field';
import '@vaadin/horizontal-layout';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('custom-field-native-input-whitespace-theme')
export class Example extends LitElement {
  protected createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-custom-field label="Coordinate" theme="whitespace">
        <vaadin-horizontal-layout theme="spacing">
          <input aria-label="Longitude" placeholder="Longitude" required type="text" />
          <input aria-label="Latitude" placeholder="Latitude" required type="text" />
        </vaadin-horizontal-layout>
      </vaadin-custom-field>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/customfield/CustomFieldWhitespaceTheme.java
package com.vaadin.demo.component.customfield;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;


public class CustomFieldWhitespaceTheme extends Div{

  public CustomFieldWhitespaceTheme() {
    // tag::snippet[]
    CoordinateField coordinate = new CoordinateField();

    coordinate.getElement().getThemeList().add("whitespace");

    add(coordinate);
    // end::snippet[]
  }
  public static class Exporter extends DemoExporter<CustomFieldWhitespaceTheme> {} // hidden-source-line
}
